samples/boards/nrf/system_off/src/retained.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <devicetree.h>
+#include <sys/byteorder.h>
+#include <sys/crc.h>
+#include <hal/nrf_power.h>
+#include "retained.h"
+
+/* nRF51 and nRF52 RAM (really, RAM AHB "slave"s) are partitioned as:
+ * * Up to 8 blocks of two 4 KiBy byte "small" sections
+ * * A 9th block of with 32 KiBy "large" sections
+ *
+ * At time of writing the maximum number of large sections is 6, all
+ * within the first large block.  Theoretically there could be more
+ * "sections" in the 9th section, and possibly more blocks.
+ */
+
+/* Inclusive address of RAM start */
+#define SRAM_BEGIN (uintptr_t)DT_REG_ADDR(DT_NODELABEL(sram0))
+
+/* Exclusive address of RAM end */
+#define SRAM_END (SRAM_BEGIN + (uintptr_t)DT_REG_SIZE(DT_NODELABEL(sram0)))
+
+/* Size of a controllable RAM section in the small blocks*/
+#define SMALL_SECTION_SIZE 4096
+
+/* Number of controllable RAM sections in each of the lower blocks */
+#define SMALL_SECTIONS_PER_BLOCK 2
+
+/* Span of a small block */
+#define SMALL_BLOCK_SIZE (SMALL_SECTIONS_PER_BLOCK * SMALL_SECTION_SIZE)
+
+/* Number of small blocks */
+#define SMALL_BLOCK_COUNT 8
+
+/* Span of the SRAM area covered by small sections */
+#define SMALL_SECTION_SPAN (SMALL_BLOCK_COUNT		\
+	* SMALL_SECTIONS_PER_BLOCK * SMALL_SECTION_SIZE)
```

Comment:
```suggestion
#define SMALL_SECTION_SPAN (SMALL_BLOCK_COUNT * SMALL_BLOCK_SIZE)
```
Wasn't `SMALL_BLOCK_SIZE` introduced for this purpose (it is not used otherwise)?